model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorage.java

```diff
@@ -0,0 +1,178 @@
+        SearchableModelField<?> field = orderBy.getModelField();
+        String modelFieldName = IckleQueryMapModelCriteriaBuilder.getFieldName(field);
+        String orderString = orderBy.getOrder().equals(QueryParameters.Order.ASCENDING) ? "ASC" : "DESC";
+
+        return modelFieldName + " " + orderString;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Stream<V> read(QueryParameters<M> queryParameters) {
+        IckleQueryMapModelCriteriaBuilder<K, V, M> iqmcb = queryParameters.getModelCriteriaBuilder().unwrap(IckleQueryMapModelCriteriaBuilder.class);
+        String queryString = iqmcb.getIckleQuery();
+
+        if (!queryParameters.getOrderBy().isEmpty()) {
+            queryString += " ORDER BY " + queryParameters.getOrderBy().stream().map(HotRodMapStorage::toOrderString)
+                                            .collect(Collectors.joining(", "));
+        }
+
+        LOG.tracef("Executing read Ickle query: %s", queryString);
+
+        QueryFactory queryFactory = Search.getQueryFactory(remoteCache);
+
+        Query<V> query = paginateQuery(queryFactory.create(queryString), queryParameters.getOffset(),
+                queryParameters.getLimit());
+
+        query.setParameters(iqmcb.getParameters());
+
+        CloseableIterator<V> iterator = query.iterator();
+        return closing(StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator, 0), false))
+                .onClose(iterator::close);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public long getCount(QueryParameters<M> queryParameters) {
+        IckleQueryMapModelCriteriaBuilder<K, V, M> iqmcb = queryParameters.getModelCriteriaBuilder().unwrap(IckleQueryMapModelCriteriaBuilder.class);
+        String queryString = iqmcb.getIckleQuery();
+
+        if (!queryParameters.getOrderBy().isEmpty()) {
+            queryString += " ORDER BY " + queryParameters.getOrderBy().stream().map(HotRodMapStorage::toOrderString)
+                    .collect(Collectors.joining(", "));
+        }
+
+        LOG.tracef("Executing count Ickle query: %s", queryString);
+
+        QueryFactory queryFactory = Search.getQueryFactory(remoteCache);
+
+        Query<V> query = paginateQuery(queryFactory.create(queryString), queryParameters.getOffset(),
+                queryParameters.getLimit());
+
+        query.setParameters(iqmcb.getParameters());
+
+        return query.execute().hitCount().orElse(0);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public long delete(QueryParameters<M> queryParameters) {
+        IckleQueryMapModelCriteriaBuilder<K, V, M> iqmcb = queryParameters.getModelCriteriaBuilder().unwrap(IckleQueryMapModelCriteriaBuilder.class);
+        String queryString = "SELECT id " + iqmcb.getIckleQuery();
+
+        if (!queryParameters.getOrderBy().isEmpty()) {
+            queryString += " ORDER BY " + queryParameters.getOrderBy().stream().map(HotRodMapStorage::toOrderString)
```

Comment:
Is ordering required with delete operation?

Reply:
I think should do it. In the future, we may want to have some default value for `maxResult` to prevent the removal of too many items and we need to remove always the same values. We are sorting also in [CHM](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/storage/chm/ConcurrentHashMapStorage.java#L145).